pyshelf/configure.py

```diff
@@ -9,20 +9,26 @@ def app(app, config_path):
         content = f.read()
         config = yaml.load(content)
 
+    if not config.get("buckets"):
+        raise ValueError("config.yaml requires buckets heading for list of buckets with associated keys.")
+
     for key, val in config.get("buckets").iteritems():
         required = {
             "accessKey": val.get("accessKey"),
             "secretKey": val.get("secretKey")
         }
+        _validate_dict(required)
+
+    required = {"elasticSearchConnectionString": config.get("elasticSearchConnectionString")}
+    _validate_dict(required)
 
+    app.config.update(config)
 
-    required["elasticSearchHost"] = config.get("elasticSearchHost")
 
+def _validate_dict(required):
     if not all(required.values()):
         raise ValueError("config.yaml did not have all required settings: " + ", ".join(required.keys()))
```

Comment:
This is going to give strange feedback if the property being validated is not that the top level.  The validation log is started to get kind of complicated.  Can we use json schema for this?  This one I would like on a separate branch though.  Lets make it off develop later